docs/DeveloperGuide.md

```diff
@@ -158,103 +158,260 @@ Classes used by multiple components are in the `seedu.address.commons` package.
 
-<puml src="diagrams/UndoSequenceDiagram-Logic.puml" alt="UndoSequenceDiagram-Logic" />
+#### Key Components
+- `DeleteCommand`: Executes the action of deleting an existing Udder based on user input. This command processes the input parameters to identify the Person object to be removed from FindingbrUdders.
+- `AddCommandParser`: Parses user input to create a `DeleteCommand` object.
+- `LogicManager`: Invokes the `DeleteCommand` to execute the deletion of an Udder.
+- `ModelManager`: Implements the `Model` interface and contains the internal list of persons (Udders).
+- `Person`: Represents an Udder in FindingbrUdders, encapsulating their personal information.
+- `AddressBookParser`: Creates a `DeleteCommand` object based on the user input.
 
-<box type="info" seamless>
+### **Sequence of action**
 
-**Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+To help you understand how the `delete` command works, here is a list of steps illustrating what occurs when [`LogicManager#execute()` is invoked](#logic-component):
 
-</box>
+We will be using the user input `delete 1` as an example.
 
-Similarly, how an undo operation goes through the `Model` component is shown below:
+1. The user inputs the command `delete 1`, intending to delete a person with index 1 in the contact list.
+2. The `DeleteCommandParser` interprets the input.
+3. A `DeleteCommand` object is created.
+4. The `LogicManager` invokes the execute method of DeleteCommand.
+5. The execute method of `DeleteCommand` invokes the `deletePerson` method in `Model` property to delete the specified `Person` object.
+6. The execute method of `DeleteCommand` returns a `CommandResult` object which stores the data regarding the completion of the `DeleteCommand`.
+7. The UI reflects this new list without the deleted `Person`.
 
-<puml src="diagrams/UndoSequenceDiagram-Model.puml" alt="UndoSequenceDiagram-Model" />
+<div markdown="span" class="alert alert-info">:information_source: **Note**:
 
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+- At step 2, if input is detected as invalid, an error will be shown on the screen and the `DeleteCommand` object will NOT be created!
+</div>
 
-<box type="info" seamless>
+---
+### Edit an Udder Feature
 
-**Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+#### **Feature**
 
-</box>
+`edit INDEX [n/NAME] [p/PHONE NUMBER] [e/EMAIL ADDRESS] [a/ADDRESS] [r/ROLE] [m/MAJOR] [t/TAG]`,
```

Comment:
I think it's clearer to write 'where INDEX is compulsory and at least one of the optional fields must be given.'